sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/newparser/NewXmlParser.java

```diff
@@ -70,11 +71,24 @@ public NewXmlParser(NewXmlFile xmlFile) throws ParserConfigurationException {
       parseXmlDeclaration();
       parseXml();
 
-    } catch (XMLStreamException|SAXException|IOException e) {
+      setDocumentLocation(xmlFile);
+
+    } catch (XMLStreamException|SAXException|IOException|ParserConfigurationException e) {
       throw new ParseException(e);
     }
   }
 
+  private static void setDocumentLocation(NewXmlFile xmlFile) {
+    Document document = xmlFile.getDocument();
+    XmlTextRange startRange = NewXmlFile.nodeLocation(document.getFirstChild());
+    XmlTextRange end = NewXmlFile.nodeLocation(document.getLastChild());
+    Optional<PrologElement> prologElement = xmlFile.getPrologElement();
+    if (prologElement.isPresent()) {
+      startRange = prologElement.get().getPrologStartLocation();
+    }
+    document.setUserData(Location.NODE.name(), new XmlTextRange(startRange, end), null);
```

Comment:
why not calling `setLocation(document, Location.NODE, startRange, end)` ?
`document` is not a Node?

Reply:
problem in `startRange`, `setLocation` requires `XmlLocation` and not `XmlTextRange`